websocket/app/server.py

```diff
@@ -0,0 +1,187 @@
+import asyncio
+import json
+import os
+import signal
+
+import aiohttp
+from aiohttp import web
+
+from modules.logger import get_logger
+from modules.ws_status import handle_close_code
+
+from websockets import serve
+
+
+# Link to API of stackmon status dashboard
+API_LINK = os.getenv("API_LINK", "http://127.0.0.1:5000/api/v1/incidents")
+# TIMEOUT is timeout for checking changes (in seconds)
+TIMEOUT = int(os.getenv("TIMEOUT", 5))
+REQUEST_TIMEOUT = int(os.getenv("REQUEST_TIMEOUT", 3))
+log_level = os.getenv("LOGLEVEL", "INFO")
+# Setup logging
+logger = get_logger(log_level)
+
+
+class SignalHandler:
+    KEEP_PROCESSING = True
+
+    def __init__(self):
+        signal.signal(signal.SIGINT, self.exit_gracefully)
+        signal.signal(signal.SIGTERM, self.exit_gracefully)
+
+    def exit_gracefully(self, signum, frame):
+        logger.info("Exiting gracefully")
+        self.KEEP_PROCESSING = False
+
+
+class DataProcessor:
+    def __init__(self):
+        self.api_link = API_LINK
+        self.previous_data = []
+        self.is_first_iteration = True
+        self.connected_clients = set()
+
+    async def check_api_availability(self):
+        try:
+            async with (
+                aiohttp.ClientSession() as session,
+                session.get(
```

Comment:
this way of fetching may be working good for the short life period. Once there are 100 entries in the DB this would be very inefficient. Instead I would propose to implement "date" query param to the incident API and with every iteration pass date of last check to get only changed content from the SDB